.github/workflows/build-and-test.yml

```diff
@@ -15,75 +15,64 @@ on:
-    with:
-      runsonlabels: '["macOS", "self-hosted"]'
-      scheme: Spezi-Package
-      destination: 'platform=tvOS Simulator,name=Apple TV 4K (3rd generation)'
-      resultBundle: Spezi-Package-tvOS.xcresult
-      artifactname: Spezi-Package-tvOS.xcresult
-  buildandtest_macos:
-    name: Build and Test Swift Package macOS
-    uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
+  package_tests:
+    name: Build and Test Swift Package ${{ matrix.platform.name }} (${{ matrix.config }})
+    uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@lukas/mac-catalyst
+    strategy:
+      matrix:
+        config: [Debug, Release]
+        platform:
+          - name: iOS
+            destination: 'platform=iOS Simulator,name=iPhone 16 Pro'
+          - name: macOS
+            destination: 'platform=macOS,arch=arm64'
+          - name: 'Mac Catalyst'
+            destination: 'platform=macOS,arch=arm64,variant=Mac Catalyst'
+          - name: watchOS
+            destination: 'platform=watchOS Simulator,name=Apple Watch Series 10 (46mm)'
+          - name: visionOS
+            destination: 'platform=visionOS Simulator,name=Apple Vision Pro'
+          - name: tvOS
+            destination: 'platform=tvOS Simulator,name=Apple TV 4K (3rd generation)'
+      fail-fast: false
     with:
       runsonlabels: '["macOS", "self-hosted"]'
       scheme: Spezi-Package
-      destination: 'platform=macOS,arch=arm64'
-      resultBundle: Spezi-Package-macOS.xcresult
-      artifactname: Spezi-Package-macOS.xcresult
-  buildandtestuitests_ios:
-    name: Build and Test UI Tests iOS
-    uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      runsonlabels: '["macOS", "self-hosted"]'
-      path: Tests/UITests
-      scheme: TestApp
-      resultBundle: TestApp-iOS.xcresult
-      artifactname: TestApp-iOS.xcresult
-  buildandtestuitests_visionos:
-    name: Build and Test UI Tests visionOS
-    uses: StanfordSpezi/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
+      destination: ${{ matrix.platform.destination }}
+      buildConfig: ${{ matrix.config }}
+      resultBundle: ${{ format('Spezi-Package-{0}-{1}.xcresult', matrix.platform.name, matrix.config) }}
+      artifactname: ${{ format('Spezi-Package-{0}-{1}.xcresult', matrix.platform.name, matrix.config) }}
+  ui_tests:
+    name: Build and Test UI Tests ${{ matrix.platform.name }} (${{ matrix.config }})
+    uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@lukas/mac-catalyst
+    strategy:
+      matrix:
+        config: [Debug, Release]
+        platform:
+          - name: iOS
+            destination: 'platform=iOS Simulator,name=iPhone 16 Pro'
+          - name: visionOS
+            destination: 'platform=visionOS Simulator,name=Apple Vision Pro'
+          # - name: macOS
+          #   destination: 'platform=macOS,arch=arm64'
```

Comment:
Probably an issue with our runner setup that this doesn't work here?

Reply:
haven't looked into this yet, but the old CI setup didn't have macOS UI testing enabled that's why i commented it out for the time being (will get resolved before merging)